pm/recipient.go

```diff
@@ -363,16 +361,21 @@ func (r *recipient) updateSenderNonce(rand *big.Int, ticket *Ticket) error {
 	defer r.senderNoncesLock.Unlock()
 
 	randStr := rand.String()
-	sn, ok := r.senderNonces[randStr]
-	if ok && ticket.SenderNonce <= sn.nonce {
-		return errors.Errorf("invalid ticket senderNonce sender=%v nonce=%v highest=%v", ticket.Sender.Hex(), ticket.SenderNonce, sn.nonce)
+	nonceToExpBlock, randKeySeen := r.senderNonces[randStr]
+	if randKeySeen {
+		_, nonceSeen := nonceToExpBlock.Get(ticket.SenderNonce)
+		if nonceSeen {
+			return errors.Errorf("invalid ticket senderNonce: already seen sender=%v nonce=%v", ticket.Sender.Hex(), ticket.SenderNonce)
+		}
+	} else {
+		r.senderNonces[randStr] = orderedmap.NewOrderedMap()
+	}
+	// add new nonce
+	r.senderNonces[randStr].Set(ticket.SenderNonce, ticket.ParamsExpirationBlock)
```

Comment:
I don't think we need to track an expiration block per senderNonce - we only need to track an expiration block per recipientRandHash (aka `randStr` in this function). And we only need to track seen senderNonce values associated with a particular recipientRandHash until that recipientRandHash expires.

I think this would simplify the state tracking since we can track an expiration block per recipientRandHash instead of an expiration block for every senderNonce value. With the current impl I think the map ends up looking like `{ key: 1 value: X, key: 2 value: X, key 3 value: X }`.